packages/core/src/utils/userop.ts

```diff
@@ -120,3 +148,50 @@ export function applyUserOpOverrideOrFeeOption(
     ? applyUserOpOverride(value, override)!
     : applyUserOpFeeOption(value, feeOption);
 }
+
+/**
+ * Utility method for checking whether the middleware pipeline should
+ * bypass the paymaster middleware for the user operation with the given overrides
+ *
+ * @template EntryPointVersion TEntryPointVersion
+ * @param overrides the user operation overrides to check
+ * @returns whether the paymaster middleware should be bypassed
+ */
+export const bypassPaymasterAndData = <
+  TEntryPointVersion extends EntryPointVersion
+>(
+  overrides: UserOperationOverrides<TEntryPointVersion> | undefined
+): boolean =>
+  !!overrides &&
+  (("paymasterAndData" in overrides && overrides.paymasterAndData === "0x") ||
+    ("paymasterData" in overrides && overrides.paymasterData === "0x"));
+
+/**
+ * Utility method for parsing the paymaster address and paymasterData from the paymasterAndData hex string
+ *
+ * @param paymasterAndData the paymaster and data hex string to parse.
+ *                         The hex string refers to the paymasterAndData field of entrypoint v0.6 user operation request
+ * @returns the parsed paymaster and paymasterData fields of entrypoint v0.7 user operation request paymaster and paymasterData field
+ */
+export const parsePaymasterAndData = (
+  paymasterAndData: Hex
+): Pick<UserOperationRequest<"0.7.0">, "paymaster" | "paymasterData"> => ({
+  paymaster: `0x${paymasterAndData.substring(2, 42)}` as Address,
+  paymasterData: `0x${paymasterAndData.substring(42)}` as Hex,
+});
+
+/**
+ * Utility method for converting the object containing the paymaster address and paymaster data
+ * to the paymaster and data concatenated hex string
+ *
+ * @param paymasterAndData the object containing the picked paymaster and paymasterData fields of
+ *                         entrypoint v0.7 user operation request
+ * @param paymasterAndData.paymaster the paymaster address
+ * @param paymasterAndData.paymasterData the paymaster data
+ * @returns the paymasterAndData hex value of entrypoint v0.6 user operation request paymasterAndData field
+ */
+export const concatPaymasterAndData = ({
+  paymaster,
+  paymasterData,
+}: Pick<UserOperationRequest<"0.7.0">, "paymaster" | "paymasterData">): Hex =>
+  `${paymaster}${paymasterData}` as Hex;
```

Comment:
This is going to end up with an "0x" in the middle, right? I'd suggest using  [`concat` from viem](https://viem.sh/docs/utilities/concat) and writing
```ts
concat(paymaster, paymasterData)
```